src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php

```diff
@@ -49,35 +51,49 @@ public function __construct(
     /**
      * @param bool $allowMultipleAttributes Whether to allow passing multiple values to the $attributes array
      */
-    public function decide(TokenInterface $token, array $attributes, mixed $object = null, bool $allowMultipleAttributes = false): bool
+    public function decide(TokenInterface $token, array $attributes, mixed $object = null, bool|AccessDecision $accessDecision = new AccessDecision(), bool $allowMultipleAttributes = false): bool
     {
+        if (\is_bool($accessDecision)) {
+            $allowMultipleAttributes = $accessDecision;
+            $accessDecision = new AccessDecision();
+        }
+
         // Special case for AccessListener, do not remove the right side of the condition before 6.0
         if (\count($attributes) > 1 && !$allowMultipleAttributes) {
             throw new InvalidArgumentException(\sprintf('Passing more than one Security attribute to "%s()" is not supported.', __METHOD__));
         }
 
-        return $this->strategy->decide(
-            $this->collectResults($token, $attributes, $object)
+        $accessDecision->strategy = $this->strategy instanceof \Stringable ? $this->strategy : get_debug_type($this->strategy);
+
+        return $accessDecision->isGranted = $this->strategy->decide(
+            $this->collectResults($token, $attributes, $object, $accessDecision)
         );
     }
 
     /**
-     * @return \Traversable<int, int>
+     * @return \Traversable<VoterInterface::ACCESS_*>
```

Comment:
Traversable requires TKey to be provided

```suggestion
     * @return \Traversable<int, VoterInterface::ACCESS_*>
```